components/file_preview.jsx

```diff
@@ -94,7 +77,7 @@ export default class FilePreview extends React.PureComponent {
 
             previews.push(
                 <div
-                    key={info.id}
+                    key={info.id + idx}
```

Comment:
Adding index to key is intentional to immediately dispose the rendered element whenever the order of files uploaded have changed.  This is for the file previews to appear in order depending on "create_at" or "name".

Reply:
Can you expand on this? Why isn't `info.id` sufficient to allow React to reorder the elements instead of having to dispose the rendered element?

Reply:
When the key is just `info.id`:
Say for example the 3rd file is successfully uploaded first, the file preview renders it on the leftmost and then next to it are the rest of uploading files.  Next, when another file (say, 2nd file) is uploaded successfully, the 3rd file will remain on the leftmost, then followed by 2nd file and then the rest of uploading files.  However, the correct order should be 2nd file | 3rd file | other uploading files.  Since the key is present & same and no change on its fileInfo, it will not be updated.

Adding `index` to `info.id`, makes it quick to dispose and render new element whenever the order changes.

Reply:
I'm not particularly sure why it's not handled by React, but at least that's what I observed so far.  Let me revisit this again.

Reply:
I don't think that's how it works: https://reactjs.org/docs/lists-and-keys.html
> We don’t recommend using indexes for keys if the order of items may change.

Reply:
I've checked this again and I stand corrected.  The observation I had before might be due to while I'm working with incomplete sorting of files.